app/components/primer/tab_nav_component.rb

```diff
@@ -119,7 +120,8 @@ class TabNavComponent < Primer::Component
     #   <% end %>
     #
     # @param label [String] Used to set the `aria-label` on the top level `<nav>` element.
-    # @param with_panel [Boolean] Whether the TabNav should navigate through pages or panels.
+    # @param with_panel [Boolean] Whether the TabNav should navigate through pages or panels. When true, <%= link_to_component(Primer::TabContainerComponent) %>
+    #   which comes with JavaScript behavior.
```

Comment:
From the docs alone, it's currently hard to know that setting `with_panel` comes with associated JS behavior. Took me a bit to trace down in the code so making docs more descriptive.